packages/js/email-editor/src/components/sidebar/details-panel.tsx

```diff
@@ -1,92 +1,22 @@
 /**
  * External dependencies
  */
-import { ExternalLink, PanelBody } from '@wordpress/components';
-import { useEntityProp } from '@wordpress/core-data';
+import { PanelBody } from '@wordpress/components';
 import { __ } from '@wordpress/i18n';
-import { createInterpolateElement } from '@wordpress/element';
-import classnames from 'classnames';
+import { applyFilters } from '@wordpress/hooks';
 
 /**
  * Internal dependencies
  */
-import { editorCurrentPostType } from '../../store';
 import { recordEvent } from '../../events';
 import { RichTextWithButton } from '../personalization-tags/rich-text-with-button';
 
-const previewTextMaxLength = 150;
-const previewTextRecommendedLength = 80;
+const SidebarExtensionComponent = applyFilters(
+	'mailpoet_email_editor_setting_sidebar_extension_component',
+	RichTextWithButton
+) as JSX.Element;
```

Comment:
If we change the type here from `JSX.Element` to `() => JSX.Element`, we can remove the ts-ignore below.

Reply:
That works. Will do. Thanks 🙏🏿

Reply:
Done in https://github.com/mailpoet/mailpoet/pull/6089/commits/6b4dc8cd95c9a0fa19d3f0a0852b3cdffbf62c39